src/ecs/explorer/ecsServiceNode.ts

```diff
@@ -38,9 +38,13 @@ export class EcsServiceNode extends AWSTreeNodeBase implements AWSResourceNode {
     public async getChildren(): Promise<AWSTreeNodeBase[]> {
         return await makeChildrenNodes({
             getChildNodes: async () => {
-                const containerNames = await this.ecs.getContainerNames(this.service.taskDefinition!)
-                const childNodes = containerNames.map(
-                    name => new EcsContainerNode(name, this.name, this.parent.arn, this.ecs, this)
+                const taskDefinition = await this.ecs.describeTaskDefinition(this.service.taskDefinition!)
+                const containers = taskDefinition.taskDefinition?.containerDefinitions
+                if (!containers) {
+                    return []
+                }
+                const childNodes = containers.map(
+                    c => new EcsContainerNode(c.name!, this.ecs, this, taskDefinition.taskDefinition?.taskRoleArn)
```

Comment:
If you're going to pass all of `this`, you may as well pull `this.ecs` from the parent at the child level as well. Either that or make `ecs` private on both sides and pass it in (may be safer but unsure how else `ecs` is being used by consumers).

Reply:
We mainly pass around clients as a means for DI so it's safe to share IMO, assuming we haven't added any state to the client.